tests/mpi/create_group_01.cc

```diff
@@ -0,0 +1,57 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+// check that Utilities::MPI::create_group is only collective over the group of
+// processes that actually want to create the communicator.
+
+#include "../tests.h"
+
+int main(int argc, char *argv[])
+{
+  MPI_Init (&argc, &argv);
```

Comment:
For some reason using `Utilities::MPI::MPI_InitFinalize` leads to a deadlock for the `MPI-3.0` compatible `OpenMPI` version.

Reply:
Probably, the same issue as https://github.com/cburstedde/p4est/issues/30.

Reply:
> Now that #6380 is merged would you mind putting that back in as an extra check on our sc settings?

Done.